torch/csrc/api/src/nn/modules/linear.cpp

```diff
@@ -26,28 +30,33 @@ LinearImpl::LinearImpl(const LinearOptions& options_) : options(options_) {
 }
 
 void LinearImpl::reset() {
-  weight =
-      register_parameter("weight", torch::empty({options.out(), options.in()}));
-  if (options.with_bias()) {
-    bias = register_parameter("bias", torch::empty(options.out()));
+  weight = register_parameter("weight", 
+    torch::empty({options.out_features(), options.in_features()}));
+  if (options.bias()) {
+    bias = register_parameter("bias", torch::empty(options.out_features()));
+  } else {
+    bias = register_parameter("bias", {});
```

Comment:
it would be awesome to also pass `/*requires_grad=*/false` into the `register_parameter()` function call, so that this doesn't print warning.

Reply:
Since [set_requires_grad is not implemented for Tensor](https://github.com/pytorch/pytorch/blob/master/c10/core/TensorImpl.h#L529) I cannot pass {} as argument to register_parameter() and I need to add another method like unsafe_ register_parameter which does not perform any checks and does not set_requires_grad